hashing/hashing.go

```diff
@@ -0,0 +1,57 @@
+package hashing
+
+import (
+	"fmt"
+	"hash/fnv"
+	"k8s.io/apimachinery/pkg/util/rand"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+	"reflect"
+)
+
+func ComputeHashForKubernetesObject(object interface{}) (string, error) {
```

Comment:
If it helps, we can also be more explicit about the interface here - meta/v1/Object I believe (https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#Object)  which all standard k8s objects implement.

Reply:
Hmm, I think that is only an interface for `ObjectMeta` though so it wouldn't give us a way to get the `Spec` field? It would be great to do it without an empty interface though if it were possible...